src/foraging_gui/Foraging.py

```diff
@@ -249,17 +284,28 @@ def connectSignalsSlots(self):
         self.Sessionlist.currentIndexChanged.connect(self._session_list)
         self.SessionlistSpin.textChanged.connect(self._session_list_spin)
         self.StartEphysRecording.clicked.connect(self._StartEphysRecording)
+        self.SetReference.clicked.connect(self._set_reference)
         # check the change of all of the QLineEdit, QDoubleSpinBox and QSpinBox
-        for container in [self.TrainingParameters, self.centralwidget, self.Opto_dialog]:
+        for container in [self.TrainingParameters, self.centralwidget, self.Opto_dialog,self.Metadata_dialog]:
             # Iterate over each child of the container that is a QLineEdit or QDoubleSpinBox
-            for child in container.findChildren((QtWidgets.QLineEdit,QtWidgets.QDoubleSpinBox,QtWidgets.QSpinBox)):        
+            for child in container.findChildren((QtWidgets.QLineEdit,QtWidgets.QDoubleSpinBox,QtWidgets.QSpinBox)):     
                 child.textChanged.connect(self._CheckTextChange)
         # Opto_dialog can not detect natural enter press, so returnPressed is used here. 
-        for container in [self.Opto_dialog]:
+        for container in [self.Opto_dialog,self.Metadata_dialog]:
             # Iterate over each child of the container that is a QLineEdit or QDoubleSpinBox
-            for child in container.findChildren((QtWidgets.QLineEdit)):        
+            for child in container.findChildren((QtWidgets.QLineEdit)):   
                 child.returnPressed.connect(self.keyPressEvent)
     
+    def _set_reference(self):
+        '''
+        set the reference point for lick spout position in the metadata dialog
```

Comment:
What is the reference position?
- Why do we set it only for the left lick spout? (we will soon be able to move the two lick spouts separately) 
- Why add a button to set the reference, instead just grabbing the location at the start or end of the session?

Reply:
This is one thing I hope to discuss. We need a reference point for the lick spout for generating the `reward_delivery` of the session metadata, which makes sense to me because the stage positions are meaningless to let us know the relative position of lick spouts to the mouse. So, we need the one reference position and the reference area (e.g. reference to the nose tip). The distance of the two lick spouts is fixed, so ideally, we only need to find the reference position for one of the lick spouts. We need to make changes if we need to move the two lick spouts separately.